apache2/modsecurity.c

```diff
@@ -153,85 +169,53 @@ int modsecurity_init(msc_engine *msce, apr_pool_t *mp) {
 #ifdef WITH_CURL
     curl_global_init(CURL_GLOBAL_ALL);
 #endif
+    lock_args = apr_pcalloc(mp, sizeof(struct waf_lock_args));
+
     /* Serial audit log mutext */
-    rc = apr_global_mutex_create(&msce->auditlog_lock, NULL, APR_LOCK_DEFAULT, mp);
-    if (rc != APR_SUCCESS) {
+    lock_name = apr_pstrcat(mp, "Global\\", "auditlog_lock", NULL);
+    set_lock_args(lock_args, lock_name, AUDITLOG_LOCK_ID, msc_waf_lock_owner);
+
+    msce->auditlog_lock = apr_pcalloc(mp, sizeof(struct waf_lock));
+    rc = Waf_createLock(msce->auditlog_lock, lock_args);    
+    if (Waf_lock_isError(rc)) {
         //ap_log_error(APLOG_MARK, APLOG_ERR, rv, s, "mod_security: Could not create modsec_auditlog_lock");
         //return HTTP_INTERNAL_SERVER_ERROR;
         return -1;
     }
 
 #ifdef WAF_JSON_LOGGING_ENABLE
     /* Serial wafjson log mutext */
-    rc = apr_global_mutex_create(&msce->wafjsonlog_lock, NULL, APR_LOCK_DEFAULT, mp);
-    if (rc != APR_SUCCESS) {
+    lock_name = apr_pstrcat(mp, "Global\\", "wafjsonlog_lock", NULL);
+    set_lock_args(lock_args, lock_name, WAFJSONLOG_LOCK_ID, msc_waf_lock_owner);
+
+    msce->wafjsonlog_lock = apr_pcalloc(mp, sizeof(struct waf_lock));
+    rc = Waf_createLock(msce->wafjsonlog_lock, lock_args);    
+    if (Waf_lock_isError(rc)) {
         //ap_log_error(APLOG_MARK, APLOG_ERR, rv, s, "mod_security: Could not create modsec_wafjsonlog_lock");
         //return HTTP_INTERNAL_SERVER_ERROR;
         return -1;
     }
 #endif
 
 #if !defined(MSC_TEST)
-#ifdef __SET_MUTEX_PERMS
```

Comment:
Could you add comments for why should remove the code?

Reply:
Sure